src/compiler/diagnosticMessages.json

```diff
@@ -951,6 +951,10 @@
         "category": "Error",
         "code": 1288
     },
+    "A non-null assertion may not be preceded by whitespace.": {

```

Comment:
What about something like

> Whitespace and comments are not allowed between the argument and '!' operator in a non-null assertion.

Reply:
Also shockingly, we don't use the term "whitespace" anywhere in our messages.

Reply:
> Non-null assertion operator must immediately follow its expression

I like Daniel's too though